pulsar-client-messagecrypto-bc/src/main/java/org/apache/pulsar/client/impl/crypto/MessageCryptoBc.java

```diff
@@ -525,6 +525,10 @@ private boolean decryptDataKey(String keyName, byte[] encryptedDataKey, List<Key
                 | NoSuchPaddingException | InvalidKeyException | InvalidAlgorithmParameterException e) {
             log.error("{} Failed to decrypt data key {} to decrypt messages {}", logCtx, keyName, e.getMessage());
             return false;
+        } catch (Exception e) {
```

Comment:
I think we can merge two catches into one catch.

Reply:
we can't merge `Exception` with other checked exceptions. but we can remove other catch block and just keep the `Exception` catch block because anyway, we are not doing any special handling with other checked-type exceptions.

Reply:
> but we can remove other catch block and just keep the Exception catch block

Ok, LGTM

Reply:
done